src/Modal.js

```diff
@@ -161,7 +161,11 @@ const Modal = React.createClass({
     let show = !!props.show;
     let dialog = React.Children.only(this.props.children);
 
-    let setMountNode = ref => this.mountNode = (!ref || ref.getMountNode());
+    let setMountNode = ref => {
+      if (ref !== null && typeof ref === 'object') {
```

Comment:
@jquense if you look at my investigations [here](https://github.com/craigbilner/react-modal-play) having a new ref (which isn't required) results in extra unnecessary work and was causing other side effects (which I didn't have time to re-create perfectly in a simple example). I couldn't see any benefit in spinning up a new function on every render?